server/http.go

```diff
@@ -100,6 +100,10 @@ func (p *Plugin) serveHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 		routeUserStart,
 		routeAPISubscribeWebhook:
 
+		if path == routeIncomingWebhook && instanceURL == "" {
+			break
+		}
+
```

Comment:
@sibasankarnayak Can you please explain the reasoning of this strategy? What if multiple Jira instances are installed? What webhook URL are you using to test this?

Reply:
@mickmister currenlty when we have a condition which check the instance id if not found in leagacy webhook it returns with error no instance found.

but the legacy webhook of version jira 2.4 was not having any instance in to just to backtrack the logic i added a pass for those webhook which don't have instance and a legacy webhook 

some example to webhook 2.4.
```
https://dkh-master-ha-13.test.mattermost.cloud/plugins/jira/webhook?channel=offtopic&secret=Eyg1Op41YHTRg1k8BsgEnf6MFMyWUYc9&team=dkh&updated_all=1

https://mattermost.hnglabs.com/plugins/jira/webhook?channel=1mwnjs3tubnjd8cgba5ucj4msh__c413ys18rjnkpjasbmxbxnz56c&secret=9UdcttdqbEJgluVRVcGmmupoNJFWov6h&team=dev
```

Reply:
nit: 1/5 more idiomatic as a separate, first case statement, rather than a nested if...break

Reply:
@levb are you referring for this logic ?
```
switch path {
	case routeIncomingWebhook:
		if instanceURL == "" {
			break
		}

		callbackInstanceID, err = p.ResolveWebhookInstanceURL(instanceURL)
		if err != nil {
			return respondErr(w, http.StatusInternalServerError, err)
		}
	case routeACJSON,
		routeACUserRedirectWithToken,
		routeACUserConfirm,
		routeACUserConnected,
		routeACUserDisconnected,
		routeOAuth1Complete,
		routeUserDisconnect,
		routeUserConnect,
		routeUserStart,
		routeAPISubscribeWebhook:

		callbackInstanceID, err = p.ResolveWebhookInstanceURL(instanceURL)
		if err != nil {
			return respondErr(w, http.StatusInternalServerError, err)
		}
	}
```

Reply:
Actually, I missed that the switch was already on `path`, ok to leave as is